Bolts/Common/BFTaskCancellationToken.m

```diff
@@ -0,0 +1,39 @@
+//
+//  BFTaskCancellationToken.m
+//  Bolts
+//
+//  Created by Daniel Hammond on 12/5/14.
+//  Copyright (c) 2014 Parse Inc. All rights reserved.
+//
+
+#import "BFTaskCancellationToken.h"
+
+@interface BFTaskCancellationToken ()
+
+@property (atomic, assign, readwrite, getter=isCancelled) BOOL cancelled;
+@property (nonatomic, retain) NSObject *lock;
```

Comment:
Synchronizing around a private lock object is generally preferable to `@synchronized(self)` blocks unless you specifically want to share the lock with other objects which is what happens with self, since other objects can call `@synchronized(you)`